### PR TITLE
[Component with its own class name as service type] LPS-198452 Refactor DeepPaginationConfigurationWrapper component

### PR DESCRIPTION
**Pull request by Valmir-unicap**
Ticket: https://liferay.atlassian.net/browse/LPS-198452

Related: https://github.com/rafaprax/liferay-portal/pull/1340

### CODE REVIEW

**Comment by Valmir-unicap** [comment]
ci:test:relevant

**Comment by Valmir-unicap** [comment]
ci:test:startup

**Comment by pyoo47** [comment]
@I had to abort the `relevant` and `startup` tests because they froze in CI

**Comment by pyoo47** [comment]
ci:test:relevant

**Comment by pyoo47** [comment]
ci:test:startup

**Comment by catarineaguiar** [comment]
@Valmir-unicap regression found:

```
[beanshell]      [exec] com.liferay.site.initializer.extender.internal.test.BundleSiteInitializerTest > testInitializeFromBundle FAILED
[beanshell]      [exec]     com.liferay.site.exception.InitializationException: com.liferay.portal.kernel.exception.OrganizationTypeException: Invalid organization type organization
[beanshell]      [exec]     	at com.liferay.site.initializer.extender.internal.BundleSiteInitializer.initialize(BundleSiteInitializer.java:623)
[beanshell]      [exec]     	at com.liferay.site.initializer.extender.internal.test.BundleSiteInitializerTest._test1(BundleSiteInitializerTest.java:3733)
[beanshell]      [exec]     	at com.liferay.site.initializer.extender.internal.test.BundleSiteInitializerTest.testInitializeFromBundle(BundleSiteInitializerTest.java:328)
[beanshell]      [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[beanshell]      [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[beanshell]      [exec]     	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[beanshell]      [exec]     	at java.lang.reflect.Method.invoke(Method.java:498)
[beanshell]      [exec]     	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable$3.evaluate(TestExecutorRunnable.java:344)
[beanshell]      [exec]     	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
[beanshell]      [exec]     	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
[beanshell]      [exec]     	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:90)
```
full log [here](https://testray.liferay.com/reports/production/logs/2023-11/test-1-35/test-portal-acceptance-pullrequest(master)/4940/modules-integration-mysql57-jdk8/0/4/jenkins-console.txt.gz?authuser=0)

```
[beanshell]      [exec] com.liferay.organizations.change.tracking.test.OrganizationTableReferenceDefinitionTest > testDiscardCTEntry FAILED
[beanshell]      [exec]     java.lang.ArrayIndexOutOfBoundsException: 0
[beanshell]      [exec]     	at com.liferay.portal.service.impl.OrganizationLocalServiceImpl.addOrganization(OrganizationLocalServiceImpl.java:247)
[beanshell]      [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[beanshell]      [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[beanshell]      [exec]     	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[beanshell]      [exec]     	at java.lang.reflect.Method.invoke(Method.java:498)
[beanshell]      [exec]     	at com.liferay.portal.spring.aop.AopMethodInvocationImpl.proceed(AopMethodInvocationImpl.java:41)
```
full log [here](https://testray.liferay.com/reports/production/logs/2023-11/test-1-35/test-portal-acceptance-pullrequest(master)/4940/modules-integration-mysql57-jdk8/0/0/jenkins-console.txt.gz?authuser=0)

**Comment by Valmir-unicap** [comment]
ci:test:sf

**Comment by Valmir-unicap** [comment]
ci:test:relevant

**Comment by Valmir-unicap** [comment]
ci:test:startup

**Comment by Valmir-unicap** [comment]
ci:test:sf

**Comment by Valmir-unicap** [comment]
ci:test:relevant

**Comment by Valmir-unicap** [comment]
ci:test:sf

**Comment by Valmir-unicap** [comment]
ci:test:relevant

**Comment by Valmir-unicap** [comment]
ci:test:startup

**Comment by catarineaguiar** [comment]
no regressions found

**Comment by Valmir-unicap** [comment]
ci:test:relevant

**Comment by Valmir-unicap** [comment]
ci:test:sf

File: modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/ElasticsearchIndexSearcher.java [hunk 1]
```diff
@@ -742,9 +777,7 @@ private int _validatePointInTimeKeepAliveSeconds(
 	private static final Log _log = LogFactoryUtil.getLog(
 		ElasticsearchIndexSearcher.class);
 
-	@Reference
-	private DeepPaginationConfigurationWrapper
-		_deepPaginationConfigurationWrapper;
+	private ConfigurationProvider _configurationProvider;
```
Comment (dantewang) [inline modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/ElasticsearchIndexSearcher.java:780]: @Valmir-unicap I think this should be marked with `@Reference`?

File: modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/deep/pagination/configuration/DeepPaginationConfigurationWrapper.java [hunk 2]
```diff
@@ -1,61 +0,0 @@
...
-import com.liferay.portal.search.elasticsearch7.configuration.DeepPaginationConfiguration;
-
-import org.osgi.service.component.annotations.Component;
-import org.osgi.service.component.annotations.Reference;
-
-/**
- * @author Gustavo Lima
- */
-@Component(
-	configurationPid = "com.liferay.portal.search.elasticsearch7.configuration.DeepPaginationConfiguration",
```
Comment (dantewang) [inline modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/deep/pagination/configuration/DeepPaginationConfigurationWrapper.java:20]: @Valmir-unicap After moving logic from this component to `ElasticsearchIndexSearcher`, this `configurationPid` thing is not moved. Is it confirmed to be redundant?

**Review by dantewang — CHANGES_REQUESTED** [review]
@Valmir-unicap Please see my comment. After you double check everything, please ping @GinsonRen to run an AssetPublisher test.

**Pull request closed without merging**